`git.properties`: support transitive references from shared projects (#1724)
SteeltoeOSS/Steeltoe: src/Management/src/GitProperties.Build/PackageReadme.md

>
> If you deploy by pushing your source code directly, rather than a pre-built or published output (for example with Cloud Foundry's `cf push`), be careful not to *also* exclude `git.properties` from whatever gets pushed or deployed. For Cloud Foundry, that means leaving it out of `.cfignore`. `git.properties` must stay out of Git through `.gitignore`, but it still needs to be present on disk and travel along with your source code.

## Using this package in a shared project

Installing this package as shown under [Getting started](#getting-started), whether with the `dotnet` CLI or Visual Studio's Add Package dialog, writes `PrivateAssets="all"` into the `<PackageReference>` line automatically. This stops the reference from becoming transitive, so `git.properties` generation stays local to the project you installed it in. That's the right choice for most solutions: usually only a few host apps need `git.properties`, so keeping the reference non-transitive avoids running Git commands anywhere else.

> Some solutions centralize Steeltoe registration in one shared project instead, for example a .NET Aspire `ServiceDefaults` project used by several host apps. In that case, install this package once, in the shared project, and make the reference transitive so every host app referencing it generates its own `git.properties` file.

Reviewer: ".NET Aspire" usage should be adjusted, here's a zero thought application of that idea (I'm not sure if you still want to mention .NET anywhere else in the sentence):
```suggestion
Some solutions centralize Steeltoe registration in one shared project instead, for example an Aspire `ServiceDefaults` project used by several host apps. In that case, install this package once, in the shared project, and make the reference transitive so every host app referencing it generates its own `git.properties` file.
```
